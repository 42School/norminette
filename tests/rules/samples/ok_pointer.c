result = result * 10 + *(*str)++ - '0';
func(i * a, *b, (&c));

int	main(void *toto)
{
	return ;
}
