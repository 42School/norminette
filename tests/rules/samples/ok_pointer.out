[36mok_pointer.c[0m - [32mIsAssignation[0m In "GlobalScope" from "None" line 1":
		<IDENTIFIER=result> <SPACE> <ASSIGN> <SPACE> <IDENTIFIER=result> <SPACE> <MULT> <SPACE> <CONSTANT=10> <SPACE> <PLUS> <SPACE> <MULT> <LPARENTHESIS> <MULT> <IDENTIFIER=str> <RPARENTHESIS> <INC> <SPACE> <MINUS> <SPACE> <CHAR_CONST='0'> <SEMI_COLON> <NEWLINE>
[36mok_pointer.c[0m - [32mIsFunctionCall[0m In "GlobalScope" from "None" line 2":
		<IDENTIFIER=func> <LPARENTHESIS> <IDENTIFIER=i> <SPACE> <MULT> <SPACE> <IDENTIFIER=a> <COMMA> <SPACE> <MULT> <IDENTIFIER=b> <COMMA> <SPACE> <LPARENTHESIS> <BWISE_AND> <IDENTIFIER=c> <RPARENTHESIS> <RPARENTHESIS> <SEMI_COLON> <NEWLINE>
[36mok_pointer.c[0m - [32mIsEmptyLine[0m In "GlobalScope" from "None" line 3":
		<NEWLINE>
[36mok_pointer.c[0m - [32mIsFuncDeclaration[0m In "GlobalScope" from "None" line 4":
		<INT> <TAB> <IDENTIFIER=main> <LPARENTHESIS> <VOID> <SPACE> <MULT> <IDENTIFIER=toto> <RPARENTHESIS> <NEWLINE>
[36mok_pointer.c[0m - [32mIsBlockStart[0m In "Function" from "GlobalScope" line 5":
		<LBRACE> <NEWLINE>
[36mok_pointer.c[0m - [32mIsExpressionStatement[0m In "Function" from "GlobalScope" line 6":
		<TAB> <RETURN> <SPACE> <SEMI_COLON> <NEWLINE>
[36mok_pointer.c[0m - [32mIsBlockEnd[0m In "Function" from "GlobalScope" line 7":
		<RBRACE> <NEWLINE>
ok_pointer.c: Error!
Error: INVALID_HEADER       (line:   1, col:   1):	Missing or invalid 42 header
